### PR TITLE
[gui] cleanup: remove log spam / code improvements

### PR DESCRIPTION
**Pull request by howie-f**
## Description
this pr started as a pure cleanup on #17129
i split into multiple commits for comfortable review. once everything is fine i'm ready to squash. apart from the cleanups, i'm not sure if all changes should be kept and are best practice. so this pr is surely open for discussion.

## Motivation and Context
remove no longer needed log-messages + further improvements

## How Has This Been Tested?
on top of current master. it does not change any existing behaviour

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [X] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/GUIPassword.h [hunk 1]
```diff
@@ -64,20 +65,24 @@ class CGUIPassword : public ISettingCallback
...
   bool IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES& vecSources);
-  /*! \brief Tests if the user is allowed to access the path by looking up the matching Mediasource
-   \param strPath The folder path to access
-   \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
-   \return If access is granted, returns \e true
-   */
-  bool IsMediaPathUnlocked(const std::string& strPath, const std::string& strType);
+
+  std::string getMediasourcePath() { return strMediasourcePath; }
+  void setMediasourcePath(std::string tmpMediasourcePath) { strMediasourcePath = tmpMediasourcePath; }
```
Comment (phunkyfish) [inline xbmc/GUIPassword.h:70]: Function should start with an uppercase letter

File: xbmc/guilib/GUIWindowManager.cpp [hunk 2]
```diff
@@ -773,9 +773,13 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
 
   // make sure we check mediasources from home
   if (GetActiveWindow() == WINDOW_HOME)
-    g_passwordManager.strMediasourcePath = !params.empty() ? params[0] : "";
+  {
+    g_passwordManager.setMediasourcePath(!params.empty() ? params[0] : "");
+  }
```
Comment (phunkyfish) [inline xbmc/guilib/GUIWindowManager.cpp:778]: Why add the braces?

File: xbmc/guilib/GUIWindowManager.cpp [hunk 3]
```diff
@@ -773,9 +773,13 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
...
   else
-    g_passwordManager.strMediasourcePath = "";
+  {
+    g_passwordManager.setMediasourcePath("");
+  }
```
Comment (phunkyfish) [inline xbmc/guilib/GUIWindowManager.cpp:782]: Why add the braces?

File: xbmc/GUIPassword.h [hunk 4]
```diff
@@ -64,20 +65,24 @@ class CGUIPassword : public ISettingCallback
...
   void RemoveSourceLocks();
   bool IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES& vecSources);
-  /*! \brief Tests if the user is allowed to access the path by looking up the matching Mediasource
-   \param strPath The folder path to access
-   \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
-   \return If access is granted, returns \e true
-   */
-  bool IsMediaPathUnlocked(const std::string& strPath, const std::string& strType);
+
+  std::string getMediasourcePath() { return strMediasourcePath; }
```
Comment (phunkyfish) [inline xbmc/GUIPassword.h:69]: Should this return a copy or a reference?

File: xbmc/GUIPassword.cpp [hunk 5]
```diff
@@ -549,30 +548,27 @@ bool CGUIPassword::IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES
   return false;
 }
 
-bool CGUIPassword::IsMediaPathUnlocked(const std::string& strPath, const std::string& strType)
+bool CGUIPassword::IsMediaPathUnlocked(const std::shared_ptr<CProfileManager>& profileManager,
+                                       const std::string& strType)
 {
-  if (StringUtils::StartsWithNoCase(strPath, "root") ||
-      StringUtils::StartsWithNoCase(strPath, "library://"))
+  std::string strPath = getMediasourcePath();
```
Comment (phunkyfish) [inline xbmc/GUIPassword.cpp:554]: const

Comment (phunkyfish) [reply xbmc/GUIPassword.cpp:554]: ref or copy?

**Review by phunkyfish — CHANGES_REQUESTED** [review]

File: xbmc/GUIPassword.h (hunk 1 above)
Comment (howie-f) [reply xbmc/GUIPassword.h:70]: uppercase, ok, the body should go into .cpp or can it stay in .h?

File: xbmc/guilib/GUIWindowManager.cpp (hunk 2 above)
Comment (howie-f) [reply xbmc/guilib/GUIWindowManager.cpp:778]: just readability

File: xbmc/GUIPassword.h (hunk 1 above)
Comment (phunkyfish) [reply xbmc/GUIPassword.h:70]: The body is fine. Simple get and set for class members is common in headers.

File: xbmc/GUIPassword.h (hunk 4 above)
Comment (howie-f) [reply xbmc/GUIPassword.h:69]: good question ;) i don't know. but i'd say copy if someday the value needs changing in that getter

Comment (ksooo) [reply xbmc/GUIPassword.h:69]: `const std::string&`, please. If somebody needs to modify the return value, they can create their own copy.

Comment (ksooo) [reply xbmc/GUIPassword.h:69]: And the method itself should be `const` and method names should start with a capital letter.
=> `const std::string& GetMediasourcePath() const { return strMediasourcePath; }`

File: xbmc/GUIPassword.h (hunk 1 above)
Comment (ksooo) [reply xbmc/GUIPassword.h:70]: param `const std::string&`, please
metod name first letter upper case, please
line no longer than 100 chars, please

File: xbmc/GUIPassword.h [hunk 6]
```diff
@@ -64,20 +65,24 @@ class CGUIPassword : public ISettingCallback
...
 
 private:
+  /*! \brief Helper to test if the user is allowed to access the path by looking up the matching Mediasource
+   \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
+   \return If access is granted, returns \e true
+   */
+  bool IsMediaPathUnlocked(const std::shared_ptr<CProfileManager>& profileManager,
+                           const std::string& strType);
+
+  std::string strMediasourcePath;
```
Comment (ksooo) [inline xbmc/GUIPassword.h:85]: member vars prefix with "m_" => `m_strMediasourcePath`

File: xbmc/guilib/GUIWindowManager.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/guilib/GUIWindowManager.cpp:778]: +1, for readability reasons

**Comment by howie-f** [comment]
huh, thanks for input! hope i didn't miss anything

File: xbmc/GUIPassword.cpp [hunk 7]
```diff
@@ -548,13 +548,12 @@ bool CGUIPassword::IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES
...
+bool CGUIPassword::IsMediaPathUnlocked(const std::shared_ptr<CProfileManager>& profileManager,
+                                       const std::string& strType)
 {
   if (!StringUtils::StartsWithNoCase(strMediasourcePath, "root") &&
       !StringUtils::StartsWithNoCase(strMediasourcePath, "library://"))
   {
-    const std::shared_ptr<CProfileManager> profileManager =
-        CServiceBroker::GetSettingsComponent()->GetProfileManager();
     if (!g_passwordManager.bMasterUser &&
         !profileManager->GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE)
```
Comment (howie-f) [inline xbmc/GUIPassword.cpp:560]: https://jenkins.kodi.tv/job/Android-ARM/18427/clang/new/source.7fe16e94-41bf-43a3-8c15-087bc1747873/#560

not sure why this generates a new warning, it's unchanged
i'd change to `profileManager->GetMasterProfile().getLockMode() != LOCK_MODE_EVERYONE)`

File: xbmc/GUIPassword.h [hunk 8]
```diff
@@ -64,23 +65,31 @@ class CGUIPassword : public ISettingCallback
...
   bool IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES& vecSources);
-  /*! \brief Tests if the user is allowed to access the path by looking up the matching Mediasource
-   \param strPath The folder path to access
-   \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
-   \return If access is granted, returns \e true
-   */
-  bool IsMediaPathUnlocked(const std::string& strPath, const std::string& strType);
+
+  // getter
+  const std::string& GetMediasourcePath() const { return strMediasourcePath; }
```
Comment (phunkyfish) [inline xbmc/GUIPassword.h:70]: return m_strMediasourcePath;

File: xbmc/GUIPassword.h [hunk 9]
```diff
@@ -64,23 +65,31 @@ class CGUIPassword : public ISettingCallback
...
+
+  // setter
+  void SetMediasourcePath(const std::string& tmpMediasourcePath)
```
Comment (phunkyfish) [inline xbmc/GUIPassword.h:73]: Maybe just `mediasourcePath` instead of `tmpMediaSourcePath`

**Review by phunkyfish — CHANGES_REQUESTED** [review]

File: xbmc/GUIPassword.h (hunk 8 above)
Comment (howie-f) [reply xbmc/GUIPassword.h:70]: c+p once again

File: xbmc/GUIPassword.h (hunk 9 above)
Comment (howie-f) [reply xbmc/GUIPassword.h:73]: makes sense

File: xbmc/GUIPassword.cpp (hunk 5 above)
Comment (garbear) [reply xbmc/GUIPassword.cpp]:
```c++
const std::string& strPath = getMediasourcePath();
```

File: xbmc/GUIPassword.cpp [hunk 10]
```diff
@@ -445,13 +444,13 @@ bool CGUIPassword::CheckMenuLock(int iWindowID)
       break;
     case WINDOW_MUSIC_NAV:      // Music
       bCheckPW = profileManager->GetCurrentProfile().musicLocked();
-      if (!bCheckPW && strMediasourcePath != "") // check mediasource by path
-        return g_passwordManager.IsMediaPathUnlocked(strMediasourcePath, "music");
+      if (!bCheckPW && !GetMediasourcePath().empty()) // check mediasource by path
```
Comment (garbear) [inline xbmc/GUIPassword.cpp:447]: "Mediasource" is not an english word, proper camel casing looks like "MediaSource"

Comment (howie-f) [reply xbmc/GUIPassword.cpp:447]: hm, thats what i'm thinkin bout since jan. "Mediasource" looks dumb

File: xbmc/GUIPassword.cpp [hunk 11]
```diff
@@ -445,13 +444,13 @@ bool CGUIPassword::CheckMenuLock(int iWindowID)
       break;
     case WINDOW_MUSIC_NAV:      // Music
       bCheckPW = profileManager->GetCurrentProfile().musicLocked();
-      if (!bCheckPW && strMediasourcePath != "") // check mediasource by path
-        return g_passwordManager.IsMediaPathUnlocked(strMediasourcePath, "music");
+      if (!bCheckPW && !GetMediaSourcePath().empty()) // check mediasource by path
```
Comment (ksooo) [inline xbmc/GUIPassword.cpp:447]: I don't get why you have introduced the getter method, as `m_strMediaSourcePath` is never read from outside class `CGUIPassword` and you don't need to mutex protect access to this member. So, no additional encapsulation is needed. So, what is the benefit of this additional code?

File: xbmc/GUIPassword.cpp [hunk 12]
```diff
@@ -445,13 +444,13 @@ bool CGUIPassword::CheckMenuLock(int iWindowID)
...
+        return g_passwordManager.IsMediaPathUnlocked(profileManager, "music");
       break;
     case WINDOW_VIDEO_NAV:      // Video
       bCheckPW = profileManager->GetCurrentProfile().videoLocked();
-      if (!bCheckPW && strMediasourcePath != "") // check mediasource by path
-        return g_passwordManager.IsMediaPathUnlocked(strMediasourcePath, "video");
+      if (!bCheckPW && !GetMediaSourcePath().empty()) // check mediasource by path
```
Comment (ksooo) [inline xbmc/GUIPassword.cpp:452]: See my other comment regarding the sense of the new getter.

File: xbmc/GUIPassword.cpp [hunk 13]
```diff
@@ -549,30 +548,27 @@ bool CGUIPassword::IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES
   return false;
 }
 
-bool CGUIPassword::IsMediaPathUnlocked(const std::string& strPath, const std::string& strType)
+bool CGUIPassword::IsMediaPathUnlocked(const std::shared_ptr<CProfileManager>& profileManager,
+                                       const std::string& strType)
 {
-  if (StringUtils::StartsWithNoCase(strPath, "root") ||
-      StringUtils::StartsWithNoCase(strPath, "library://"))
+  const std::string& strPath = GetMediaSourcePath();
```
Comment (ksooo) [inline xbmc/GUIPassword.cpp:554]: See my other comment regarding the sense of the new getter.

File: xbmc/GUIPassword.cpp (hunk 11 above)
Comment (howie-f) [reply xbmc/GUIPassword.cpp:447]: you're right, there is no benefit. the reason is i am not an experienced c++ programmer. was just looking at my first pr ever (which would never be there without help) and trying to tune it a little. call it practicing ;)
setting a public var (now `m_strMediaSourcePath`) from outside looked somehow odd to me, too. so i thought about getter and setter.
if you say this makes it all more complicated, i have no problem with dropping it.
> i'm not sure if all changes should be kept 

(just tell me)

Comment (howie-f) [reply xbmc/GUIPassword.cpp:447]: imo 072afd22d8142556110b4a790f65ee07d2407db6 would be a valid usecase for the setter.

Comment (ksooo) [reply xbmc/GUIPassword.cpp:447]: > i am not an experienced c++ programmer. 

 You are welcome and we are happy to assist you.

> setting a public var (now m_strMediaSourcePath) from outside looked somehow odd to me (in this context). 

 And you are right with this. public member vars break encapsulation (on of the basic principles of OOP).

> so i thought about getter and setter.

 The setter is fine, for the reason given above. The getter is not needed here as there is no access from outside the class.

Comment (ksooo) [reply xbmc/GUIPassword.cpp:447]: >>   i'm not sure if all changes should be kept

> (just tell me)

 Keep the setter, remove the getter.

Comment (howie-f) [reply xbmc/GUIPassword.cpp:447]: thanks - on the way

File: xbmc/GUIPassword.h [hunk 14]
```diff
@@ -64,23 +65,27 @@ class CGUIPassword : public ISettingCallback
...
   int iMasterLockRetriesLeft;
-  std::string strMediasourcePath;
 
 private:
+  /*! \brief Helper function to test if the user is allowed to access the path
+   by looking up the matching Mediasource. Used internally by CheckMenuLock.
+   \param profileManager instance passed by ref. see CGUIPassword::CheckMenuLock
+   \param strType The type of share being accessed, e.g. "music", "video", etc.
+   \return If access is granted, returns \e true
+   */
```
Comment (howie-f) [inline xbmc/GUIPassword.h:83]: those priv declarations don't appear in the codedocs. fine if i remove the doxy?

Comment (ksooo) [reply xbmc/GUIPassword.h:83]: No, keep it. Every documentation, given content use correct, is valuable. If there are problems with doxygen, root car should be fixed. But not in this PR.

**Comment by ksooo** [comment]
By the why, for the "badges" in PR titles, we do denote the affected components, not the kind of change.

**Comment by ksooo** [comment]
Right badge for your PR would be [gui]

**Comment by howie-f** [comment]
ok, getter removed, runtime-test still fine, here we go

File: xbmc/GUIPassword.cpp [hunk 15]
```diff
@@ -549,30 +548,34 @@ bool CGUIPassword::IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES
...
-    // no mediasource-lookup needed
-    CLog::Log(LOGDEBUG, "CGUIPassword::IsMediaPathUnlocked - entering from {}", strPath);
-    return true;
+    m_strMediaSourcePath = MediaSourcePath;
   }
+  else
+  {
+    m_strMediaSourcePath.clear();
+  }
+}
```
Comment (ksooo) [inline xbmc/GUIPassword.cpp:562]: No, please keep the setter as simple and inline in the header file as it was before! No additional logics in the setter, please. keep the root/library check in CGUIPassword::IsMediaPathUnlocked as it was before, please.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/GUIPassword.cpp (hunk 15 above)
Comment (howie-f) [reply xbmc/GUIPassword.cpp:562]: > imo 072afd2 would be a valid usecase for the setter.

aaahhh, ok, thought you saw that "usecase"-commit... i'll change it back

**Review by ksooo — APPROVED** [review]
Thanks.

**Comment by howie-f** [comment]
main intention is to rid the log messages. would be happy to see them disappear in leia, too. the other stuff on top would stay out of v18, right?

jenkins build this please

**Comment by ksooo** [comment]
Only bug fixes go into Leia, sorry.

**Comment by ksooo** [comment]
Removing log messages is not a bug fix.

**Comment by howie-f** [comment]
yes - true np. think we're done here now that everythings green. thanks for your assistence 😀

**Comment by ksooo** [comment]
@phunkyfish you were requesting changes. Is it okay to shove this in in its current state?

**Review by phunkyfish — APPROVED** [review]
Yes, all good from me now.

**Review by garbear — APPROVED** [review]

File: xbmc/GUIPassword.h [hunk 16]
```diff
@@ -64,23 +65,33 @@ class CGUIPassword : public ISettingCallback
...
   void RemoveSourceLocks();
   bool IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES& vecSources);
-  /*! \brief Tests if the user is allowed to access the path by looking up the matching Mediasource
-   \param strPath The folder path to access
-   \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
-   \return If access is granted, returns \e true
-   */
-  bool IsMediaPathUnlocked(const std::string& strPath, const std::string& strType);
+
+  // getter
```
[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/GUIPassword.h]: whoops, you have an outdated file here

**Comment by garbear** [comment]
thanks for the improvement!

**Pull request merged**